### PR TITLE
Catch \Throwable

### PR DESCRIPTION
**Pull request by fprochazka**
| Q | A |
| --- | --- |
| Branch? | 2.3, 2.7, 2.8, 3.0 |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | Yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

Related symfony/symfony#18765, symfony/symfony#15949

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
:+1:

File: src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php [hunk 1]
```diff
@@ -77,6 +77,9 @@ public function load($resource, $type = null)
         } catch (\Exception $e) {
             $this->loading = false;
             throw $e;
+        } catch (\Throwable $e) {
+            $this->loading = false;
+            throw $e;
```
Comment (AyrtonRicardo) [inline src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php]: in php7 isn't possible implements ``\Throwable`` interface only extends `\Exception` class, so why not use only one catch with `\Throwable`?

Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php]: because we support PHP5 also

Comment (AyrtonRicardo) [inline src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php]: i understud

File: src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php [hunk 2]
```diff
@@ -193,8 +194,8 @@ public function toolbarAction(Request $request, $token)
         $url = null;
         try {
             $url = $this->generator->generate('_profiler', array('token' => $token));
-        } catch (\Exception $e) {
-            // the profiler is not enabled
+        } catch (RouteNotFoundException $e) {
+            // the named route doesn't exist => the profiler is not enabled
```
Comment (xabbuh) [inline src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php]: We now don't catch other exceptions anymore (which might happen when the `_profiler` route is overridden).

Comment (fprochazka) [inline src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php]: True, but does that happen?

Comment (nicolas-grekas) [inline src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php]: @fprochazka let's stay on the safe side

Comment (fprochazka) [inline src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php]: Ok, so I'll add all 3 of them? Or do I revert this for now?

Comment (nicolas-grekas) [inline src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php]: let's revert

**Comment by nicolas-grekas** [comment]
Thank you @fprochazka.

File: src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php (hunk 2 above)
Comment (patrick-mcdougle) [inline src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php]: @nicolas-grekas @fabpot It doesn't look like this change was reverted before merging... just thought you'd want to know.

Comment (fabpot) [inline src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php]: ping @nicolas-grekas You merged this one.

**Pull request closed without merging**